doc/adr/0003-directory-structure.md

```diff
@@ -0,0 +1,39 @@
+# 3. Define directory structure
+
+Date: 2020-12-01
+
+## Status
+
+Accepted
+
+## Context
+
+There will be many services contributed by multiple people. We need a common way of structuring the services
+so it's easy for the user to pick up any one service.
+
+## Decision
+
```

Comment:
I think we can split the `Decision` in several parts: `services` and `extensions`. Maybe we could mention test that goes into `tests`. I don't think there is a need to detail it at the moment.  

For extensions, what do you think of having them in an `extensions` folder organized like Mbed OS BLE ? 

- `extensions/include/ble/gatt`
- `extensions/include/ble/gap`
- `extensions/include/ble/common`
- `extensions/source/...`

That way, include path shouldn't change if these extensions are included into Mbed OS.

Reply:
sounds good, might as well put this into this adr